tests/conftest.py

```diff
@@ -197,3 +213,60 @@ def cylc_validate_cli(capsys, caplog):
 @pytest.fixture(scope='module')
 def mod_cylc_validate_cli(mod_capsys, mod_caplog):
     return _cylc_validate_cli(mod_capsys, mod_caplog)
+
+
+@pytest.fixture
+def file_poll():
+    """Poll for the existance of a file.
+    """
+    def _inner(
+        fpath: "Path", timeout: int = 5, inverse: bool = False
+    ):
+        timeout_func(
+            lambda: fpath.exists() != inverse,
+            f"file {fpath} {'still' if inverse else 'not'} found after "
+            f"{timeout} seconds",
+            timeout
+        )
+    return _inner
+
+
+@pytest.fixture
+def purge_workflow(run_ok, file_poll):
+    """Ensure workflow is stopped and cleaned"""
+    def _inner(id_, timeout=5):
+        stop = f'cylc stop {id_} --now --now'
+        clean = f'cylc clean {id_}'
+        timeout_func(
+            partial(run_ok, stop),
+            message=f'Not run after {timeout} seconds: {stop}')
+        file_poll(
+            Path.home() / 'cylc-run' / id_ / '.service/contact',
+            inverse=True,
+        )
+        timeout_func(
+            partial(run_ok, clean),
+            message=f'Not run after {timeout} seconds: {clean}')
```

Comment:
(Not passing the timeout on here)

```suggestion
            message=f'Not run after {timeout} seconds: {stop}',
            timeout
        )
        file_poll(
            Path.home() / 'cylc-run' / id_ / '.service/contact',
            inverse=True,
        )
        timeout_func(
            partial(run_ok, clean),
            message=f'Not run after {timeout} seconds: {clean}',
            timeout
        )
```